assets/js/components/Link.test.js

```diff
@@ -48,10 +48,58 @@ describe( 'Link', () => {
 			expect( container.firstChild ).toHaveAttribute( 'aria-label', 'label prop (opens in a new tab)' );
 		} );
 
-		it( 'ignores valid non-string children when building an aria-label for an external link', () => {
+		it( 'adds an aria-label with the text content for disabled links', () => {
+			const { container } = render( <Link disabled onClick={ () => {} }>text content</Link> );
+
+			expect( container.firstChild ).toHaveAttribute( 'aria-label', 'text content (disabled)' );
+		} );
+
+		it( 'uses the existing aria-label for disabled links', () => {
+			const { container } = render( <Link aria-label="label prop" disabled onClick={ () => {} }>text content</Link> );
+
+			expect( container.firstChild ).toHaveAttribute( 'aria-label', 'label prop (disabled)' );
+		} );
+
+		it( 'does not build an ARIA label for non-string children', () => {
 			const { container } = render( <Link external>{ [ 'text content' ] }</Link> );
 
-			expect( container.firstChild ).toHaveAttribute( 'aria-label', '(opens in a new tab)' );
+			expect( container.firstChild ).not.toHaveAttribute( 'aria-label' );
+		} );
+
+		it( 'does not append anything to the ARIA label when not needed', () => {
+			const { container } = render( <Link aria-label="foo">text content</Link> );
+
+			expect( container.firstChild ).toHaveAttribute( 'aria-label', 'foo' );
+		} );
+
+		it( 'does not set the ARIA label based on children when there is no suffix', () => {
+			const { container } = render( <Link>text content</Link> );
+
+			expect( container.firstChild ).not.toHaveAttribute( 'aria-label' );
+		} );
+
+		it( 'creates an <a> attribute by default', () => {
```

Comment:
I think "tag" or "element" may be more appropriate here?
```suggestion
		it( 'creates an <a> tag by default', () => {
```